### PR TITLE
PRIV-8482: "Create password" popup now has blocking design

### PR DESCRIPTION
**Pull request by yoklmnprst**

### CODE REVIEW

File: Sources/PrivoSDK/PrivoAuth.swift [hunk 1]
```diff
@@ -237,16 +237,32 @@ public class PrivoAuth {
...
+        var resetPasswordLink = response.to.updatePasswordLink
         
-        return .init(resetPasswordLink: response.to.updatePasswordLink)
+        let serviceIdentifier = PrivoInternal.settings.serviceIdentifier
+        if let serviceInfo = await api.getServiceInfo(serviceIdentifier: serviceIdentifier),
+           let siteId = serviceInfo.p2siteId
+        {
+            var components = resetPasswordLink.urlComponent()
+            let queryItemsOld = components.queryItems ?? []
+            let queryItemsNoSiteId = queryItemsOld.filter {
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/PrivoAuth.swift:248]: how about moving this logic to the extension?

File: Sources/PrivoSDK/internal/UpdatePasswordView.swift [hunk 2]
```diff
@@ -8,28 +8,64 @@ struct UpdatePasswordView: View {
...
+                .transition(.opacity)
+                .animation(.easeOut(duration: 0.3), value: isPresentedAnimation)
+            
+            VStack(spacing: 0) {
+                Spacer()
+                VStack(spacing: 0) {
+                    Rectangle()
+                        .fill(backgroundColor)
+                        .frame(maxWidth: .infinity)
+                        .frame(height: 26) // Sometimes, the background color may not be transparent, and applying a white background may not be visible. In this case, the header color will not match the background color. Let's make the header height moderate and not too narrow, so that in this case, the overall appearance of the pop-up widget does not look bad.
```
[the remark this answers is not in the document]
Comment (AlexSlo-Dev) [reply Sources/PrivoSDK/internal/UpdatePasswordView.swift:54]: please implement this dialog as we discussed

File: Sources/PrivoSDK/internal/UpdatePasswordView.swift [hunk 3]
```diff
@@ -8,28 +8,64 @@ struct UpdatePasswordView: View {
...
+                        .cornerRadiusTop(8)
+                    
+                    ModalWebView(isPresented: $isPresented, config: config)
+                        .background(backgroundColor)
+                        .frame(maxWidth: .infinity)
+                        .frame(height: 500)
+                }
+            }.edgesIgnoringSafeArea(.bottom)
+            .onAppear {
+                DispatchQueue.main.asyncAfter(deadline: .now()+0.5) {
```
Comment (AlexSlo-Dev) [inline Sources/PrivoSDK/internal/UpdatePasswordView.swift:64]: please explain me Why

File: Sources/PrivoSDK/PrivoAuth.swift (hunk 1 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/PrivoAuth.swift:248]: done

File: Sources/PrivoSDK/internal/UpdatePasswordView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (yoklmnprst) [reply Sources/PrivoSDK/internal/UpdatePasswordView.swift]: Discussed and decided to remove this header.

File: Sources/PrivoSDK/internal/UpdatePasswordView.swift (hunk 3 above)
Comment (yoklmnprst) [reply Sources/PrivoSDK/internal/UpdatePasswordView.swift:44]: This delay is associated with the overlap of two animations in UIKit (UIHostingController) and SwiftUI elements. Not an ideal solution, I understand, but I decided to do it for now.

**Review by AlexSlo-Dev — APPROVED** [review]

**Pull request merged**